mico-core/src/main/java/io/github/ust/mico/core/broker/DeploymentBroker.java

```diff
@@ -79,13 +79,15 @@ public MicoApplicationJobStatus deployApplication(String shortName, String versi
                 .setStatus(MicoServiceBackgroundJob.Status.RUNNING);
             backgroundJobBroker.saveJob(job);
 
-            Optional<MicoServiceDeploymentInfo> serviceDeploymentInfoOptional = serviceDeploymentInfoRepository
+            List<MicoServiceDeploymentInfo> serviceDeploymentInfos = serviceDeploymentInfoRepository
```

Comment:
Maybe add a helper method in here that checks if the list only has one element, throws the relevant exceptions and returns an optional.